bin/main.rb

```diff
@@ -1 +1,82 @@
-puts "Hello World!"
\ No newline at end of file
+def welcome_message
+  puts 'Welcome to Tic Tac Toe Game'
+  end
+welcome_message
+
+def user_setup
+  print 'Please Enter Player One name '
+  player1 = gets.chomp
+  print 'Please Enter Player Second name '
+  player2 = gets.chomp
+
+  print "Choose #{player1} X or O: "
+  mark = gets.chomp
+  mark.upcase!
+  if mark == 'X'
+    puts "X is assigned to #{player1} and O is assigned to #{player2}"
+  else
+    puts "O is assigned to #{player1} and X is assigned to #{player2}"
+  end
+end
+
+user_setup
+
+puts 'Game Board'
+board = [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ']
+def display_board(board)
+  puts " #{board[0]} | #{board[1]} | #{board[2]} "
+  puts '-----------'
+  puts " #{board[3]} | #{board[4]} | #{board[5]} "
+  puts '-----------'
+  puts " #{board[6]} | #{board[7]} | #{board[8]} "
+end
+display_board(board)
+
+def empty_index_board(board, index)
+  if board[index] == ' ' || board[index] == 'X' || board[index] == 'O'
```

Comment:
- [ ] Instead of checking three things you could just check if there was any letter at all inside the board cell string. You can use regular expressions for that or even easier, if you make the free board cells empty strings like '' instead of ' ', use the #empty? method.